packages/core/core/src/public/Config.js

```diff
@@ -50,12 +58,22 @@ export default class PublicConfig implements IConfig {
     return this.#config.isSource;
   }
 
-  get resolvedPath() {
-    return this.#config.resolvedPath;
+  get includedFiles() {
+    return this.#config.includedFiles;
   }
 
-  setResolvedPath(filePath: FilePath) {
-    this.#config.resolvedPath = filePath;
+  get rootDir() {
```

Comment:
Should this actually exist?

I also don't think it's calculated correctly at the moment, it should probably take the lock or pkg.json file that is closest to `/` to have a real root? The only real purpose this has is for `resolveFrom`

Reply:
I don't think it's very useful. `devDeps` should usually be resolved  from the config file in which they are defined. One option is to change `addDevDependency` to something like:

```js
addDevDependency(devDep: {|moduleSpecifier: PackageName, resolveFrom?: FilePath, version?: SemVer|})
```

If `resolveFrom` is defined, then Parcel core would take care of resolving, and if `version` were defined then that means the plugin already resolved it (would probably throw if they are both defined). Another option, which I am leading towards, is to get rid of `addDevDependency` entirely. Plugins would then have to incorporate the `devDep` versions in their result hash and add the lock file to included files themselves if necessary. Then we could get rid of the `dep_version_request` type which would make the `RequestGraph` smaller. They aren't bringing much additional benefit anyway because the package manager has it's own caching. I guess a middle ground option would be to keep `addDevDependency`, but always expect a version. Then core could take care of including it in the hash and setting up the invalidation for the lock file. Not sure how much that helps though and we would have to explain why it's necessary.

Reply:
`addDevDependency` already appears to be just a name and version at the moment?

`addDevDependency(name: PackageName, version?: string)` should anything be changed, all tests seem to pass?

Reply:
I think we should get rid of version and change it to `addDevDependency(moduleSpecifier: PackageName, resolveFrom: FilePath)`. One thing I didn't consider initially is local `devDeps`, which people expect to invalidate on source change instead of version. That way Parcel can resolve the dep and use a version number for external packages and source code hash for internal packages or relative paths.

That being said, `devDeps` aren't invalidating properly at the moment anyway if you don't pass a version as the second argument because core never ends up resolving the versions. That was a regression from a refactor, so I guess we don't have a test for that. We only have a test for Babel plugins upgrading, but the Babel transformer passes the version. So I'd say it's up to you if you want to make these changes in this PR or leave `devDeps` somewhat broken and we'll follow up in another PR.

Reply:
@padmaia it sounds like a pretty substantial change, so maybe we should do it in a seperate PR like you mentioned, I can change the function params but from what I can tell I need to change quite a bit more than just these params and as it's currently already broken it wouldn't really make a difference anyway.